connector/connect/src/main/protobuf/spark/connect/base.proto

```diff
@@ -100,18 +70,138 @@ message AnalyzePlanRequest {
   // logging purposes and will not be interpreted by the server.
   optional string client_type = 4;
 
-  // (Optional) Get the explain string of the plan.
-  Explain explain = 5;
+  repeated AnalysisTask tasks = 5;
+
+  message AnalysisTask {
```

Comment:
I think this is overcomplicating the message type. For elements that are just plain empty we should not push them to the client.